Modules/Scripted/VPAWModel/VPAWModel.py

```diff
@@ -377,58 +414,287 @@ def __init__(self):
+
+        Parameters
+        ----------
+        vPAWRootDirectory : str
+            This is the input/output directory, e.g., "path/to/vpaw-data-root".  It must
+            contain FilteredControlBlindingLogUniqueScanFiltered.xls, images/*, and
+            landmarks/*.
+        vPAWModelsDirectory : str
+            It must contain a file with name like "116(158.10-38.AM.24.Mar).pth".
         """
+        startTime = time.time()
+        logging.info("Pediatric Airway Atlas pipeline started")
+
+        # self.convertCTScansToNRRD(vPAWRootDirectory)
+        self.convertFCSVLandmarksToP3(vPAWRootDirectory)
+        self.runSegmentation(vPAWRootDirectory, vPAWModelsDirectory)
 
-        if not inputVolume or not outputVolume:
-            raise ValueError("Input or output volume is invalid")
+        stopTime = time.time()
+        logging.info(
+            f"Pediatric Airway Atlas pipeline completed in {stopTime-startTime:.2f}"
+            + " seconds"
+        )
 
-        import time
+    def convertFCSVLandmarksToP3(self, vPAWRootDirectory):
+        import conversion_utils.generate_pixel_space_landmarks
 
-        startTime = time.time()
-        logging.info("Processing started")
-
-        # Compute the thresholded output volume using the "Threshold Scalar Volume" CLI
-        # module
-        cliParams = {
-            "InputVolume": inputVolume.GetID(),
-            "OutputVolume": outputVolume.GetID(),
-            "ThresholdValue": imageThreshold,
-            "ThresholdType": "Above" if invert else "Below",
-        }
-        cliNode = slicer.cli.run(
-            slicer.modules.thresholdscalarvolume,
-            None,
-            cliParams,
-            wait_for_completion=True,
-            update_display=showResult,
+        conversion_utils.generate_pixel_space_landmarks.convert_landmarks(
+            os.path.join(vPAWRootDirectory, "images"),
+            os.path.join(vPAWRootDirectory, "landmarks"),
+            os.path.join(vPAWRootDirectory, "transformed_landmarks"),
         )
-        # We don't need the CLI module node anymore, remove it to not clutter the scene
-        # with it
-        slicer.mrmlScene.RemoveNode(cliNode)
 
-        stopTime = time.time()
-        logging.info(f"Processing completed in {stopTime-startTime:.2f} seconds")
+    def runSegmentation(self, vPAWRootDirectory, vPAWModelsDirectory):
+        cwd = os.getcwd()
+        ConfigHandle, ConfigName = None, None
+        SegmentHandle, SegmentName = None, None
+        try:
+            # Change directory and create files
+            os.chdir(self.pediatric_airway_atlas_directory)
+            ConfigHandle, ConfigName = tempfile.mkstemp(suffix=".yaml", text=True)
+            SegmentHandle, SegmentName = tempfile.mkstemp(suffix=".yaml", text=True)
+
+            # Add text to the main configuration file
+            b_s_f_s = "False"
+            n_p = 1
+            os.write(
```

Comment:
It would be better to read and write yaml using pyyaml.

This would make it easier to extend configurability later, which I imagine is something we will want to do.

Reply:
Good thought.  Will do.